runtime/common/src/lib.rs

```diff
@@ -390,59 +394,88 @@ pub mod investment_portfolios {
 		investor: <T as frame_system::Config>::AccountId,
 	) -> Vec<(
 		<T as pallet_investments::Config>::InvestmentId,
-		InvestmentPortfolio<Balance>,
+		InvestmentPortfolio<Balance, CurrencyId>,
 	)>
 	where
-		T: frame_system::Config + pallet_investments::Config + orml_tokens::Config,
-		<T as pallet_investments::Config>::InvestmentId:
-			TrancheCurrency<PoolId, TrancheId> + Into<<T as orml_tokens::Config>::CurrencyId> + Ord,
-		CurrencyId: From<<T as orml_tokens::Config>::CurrencyId>,
+		T: frame_system::Config
+			+ pallet_investments::Config
+			+ orml_tokens::Config
+			+ pallet_restricted_tokens::Config,
+		<T as pallet_investments::Config>::InvestmentId: TrancheCurrency<PoolId, TrancheId>
+			+ Into<<T as orml_tokens::Config>::CurrencyId>
+			+ Ord
+			+ Into<<T as pallet_restricted_tokens::Config>::CurrencyId>,
+		CurrencyId: From<<T as orml_tokens::Config>::CurrencyId>
+			+ From<<T as pallet_restricted_tokens::Config>::CurrencyId>,
+		<T as pallet_restricted_tokens::Config>::CurrencyId:
+			From<<T as orml_tokens::Config>::CurrencyId>,
 		Balance: From<<T as pallet_investments::Config>::Amount>
-			+ From<<T as orml_tokens::Config>::Balance>,
+			+ From<<T as pallet_restricted_tokens::Config>::Balance>,
 		PoolInspector: PoolInspect<
 			<T as frame_system::Config>::AccountId,
-			<T as orml_tokens::Config>::CurrencyId,
+			<T as pallet_restricted_tokens::Config>::CurrencyId,
 			PoolId = PoolId,
 			TrancheId = TrancheId,
 			Moment = Seconds,
 		>,
 	{
 		let mut portfolio = BTreeMap::<
 			<T as pallet_investments::Config>::InvestmentId,
-			InvestmentPortfolio<Balance>,
+			InvestmentPortfolio<Balance, CurrencyId>,
 		>::new();
 
 		// Denote current tranche token balances before dry running collecting
 		orml_tokens::Accounts::<T>::iter_key_prefix(&investor).for_each(|currency| {
 			if let CurrencyId::Tranche(pool_id, tranche_id) = CurrencyId::from(currency) {
-				let balance = orml_tokens::Accounts::<T>::get(&investor, currency);
+				let pool_currency = PoolInspector::currency_for(pool_id)
+					.expect("Pool must exist; qed")
```

Comment:
I think we can safely expect here. If we want to be super duper safe, I can wrap the rest of the inner scopes into a `.map`. WDYT @mustermeiszer ?

Reply:
Yeah, I would also say so. I also think zo remember that api calls are wrapped in a panic catching-something, I will double check that. 

But think the expect is fine!